src/device.rs

```diff
@@ -535,3 +581,35 @@ pub extern "C" fn wgpu_buffer_get_mapped_range(
 ) -> *mut u8 {
     gfx_select!(buffer_id => GLOBAL.buffer_get_mapped_range(buffer_id, start, size))
 }
+
+/// Fills the given `info` struct with the adapter info.
+///
+/// # Safety
+///
+/// The field `info.name` is expected to point to a pre-allocated memory
+/// location. This function is unsafe as there is no guarantee that the
+/// pointer is valid and big enough to hold the adapter name.
+#[no_mangle]
+pub unsafe extern "C" fn wgpu_adapter_get_info(adapter_id: id::AdapterId, info: &mut CAdapterInfo) {
+    let adapter_info = gfx_select!(adapter_id => GLOBAL.adapter_get_info(adapter_id));
+    let adapter_name = CString::new(adapter_info.name).unwrap();
+
+    info.device = adapter_info.device;
+    info.vendor = adapter_info.vendor;
+    info.device_type = CDeviceType::from(adapter_info.device_type);
+    info.backend = adapter_info.backend;
+
+    let string_bytes = adapter_name.as_bytes_with_nul();
+    let cpy_length = std::cmp::min(info.name_length, string_bytes.len());
+
+    // Copies the string bytes owned into a the pre-allocated memory location
+    // pointed by `info.name`.
+    // NOTE: this is obviousy unsafe and the caller **must** ensure the
+    // memory is allocated.
+    ptr::copy(string_bytes.as_ptr(), info.name as *mut u8, cpy_length - 1);
```

Comment:
something needs to bail out if `cpy_length` is 0. Rust doesn't do underflow checks in Release.

Reply:
I wasn't sure :D thanks for the precision!